integration flake: TestUnschedulableNodes is flaky

```
panic: interface conversion: interface {} is nil, not *api.Node [recovered]
	panic: interface conversion: interface {} is nil, not *api.Node

goroutine 42845 [running]:
panic(0x24413e0, 0xc8383133c0)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
testing.tRunner.func1(0xc83bebbc20)
	/usr/local/go/src/testing/testing.go:467 +0x192
panic(0x24413e0, 0xc8383133c0)
	/usr/local/go/src/runtime/panic.go:443 +0x4e9
k8s.io/kubernetes/test/integration.waitForReflection.func1(0x1deefa0, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/integration/scheduler_test.go:112 +0x222
k8s.io/kubernetes/pkg/util/wait.WaitFor(0xc837aa12a0, 0xc838313300, 0xc838175320, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/wait/wait.go:210 +0x77
k8s.io/kubernetes/pkg/util/wait.pollInternal(0xc837aa12a0, 0xc838313300, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/wait/wait.go:170 +0x8a
k8s.io/kubernetes/pkg/util/wait.Poll(0x5f5e100, 0x6fc23ac00, 0xc838313300, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/pkg/util/wait/wait.go:164 +0x4f
k8s.io/kubernetes/test/integration.waitForReflection(0xc83bebbc20, 0x7f210f5a71e8, 0xc837c740a0, 0x2a3dce0, 0x19, 0x2e00598, 0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/integration/scheduler_test.go:108 +0x172
k8s.io/kubernetes/test/integration.DoTestUnschedulableNodes.func1(0xc83bebbc20, 0xc831b12500, 0x7f210f5a71e8, 0xc837c740a0, 0xc838c9ae80)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/integration/scheduler_test.go:178 +0x249
k8s.io/kubernetes/test/integration.DoTestUnschedulableNodes(0xc83bebbc20, 0xc838c9ae80, 0x7f210f5a71e8, 0xc837c740a0)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/integration/scheduler_test.go:242 +0xaf9
k8s.io/kubernetes/test/integration.TestUnschedulableNodes(0xc83bebbc20)
	/go/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/integration/scheduler_test.go:84 +0x73e
testing.tRunner(0xc83bebbc20, 0x3b6ee50)
	/usr/local/go/src/testing/testing.go:473 +0x98
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:582 +0x892
```

@davidopp @mml

--- a/test/integration/scheduler_test.go
+++ b/test/integration/scheduler_test.go
@@ -109,7 +109,15 @@ func waitForReflection(t *testing.T, s cache.Store, key string, passFunc func(n 
 		if n, _, err := s.GetByKey(key); err == nil && passFunc(n) {
 			return true, nil
 		} else {
-			nodes = append(nodes, n.(*api.Node))
+			if err != nil {
+				t.Errorf("Unexpected error: %v", err)
+			} else {
+				if n == nil {
+					nodes = append(nodes, nil)
+				} else {
+					nodes = append(nodes, n.(*api.Node))
+				}
+			}
 			return false, nil
 		}
 	})

